packages/core/src/new-api/internal/validation/futures/validateNamedContractDeployment.ts

```diff
@@ -0,0 +1,30 @@
+import { ethers } from "ethers";
+
+import { IgnitionValidationError } from "../../../../errors";
+import { isArtifactType } from "../../../type-guards";
+import { ArtifactResolver } from "../../../types/artifact";
+import { NamedContractDeploymentFuture } from "../../../types/module";
+
+export async function validateNamedContractDeployment(
+  future: NamedContractDeploymentFuture<string>,
+  artifactLoader: ArtifactResolver
+) {
+  const artifact = await artifactLoader.load(future.contractName);
```

Comment:
Should we handle the case were the name doesn't exist? Or maybe just let the adapter throw an error, letting it do whatever it wants.

Reply:
In this specific case, our adapter (i.e. `hre`) does throw an error. I'm on board with continuing to delegate that responsibility to the adapter